### DISCUSSION TITLE
How to Request rate limiting with fraction/custom time unit?

### DISCUSSION THREAD

**Discussion by GlassGruber** [curl/curl#14242] [Q&A] [answered]
I naively tested to rate limit curl with `--rate 5/10s` but complained with following message:

```
curl: unsupported --rate unit
curl: conf:1: '--rate' is badly used here
curl: cannot read config from 'conf'
curl: option --config: error encountered when reading a file
curl: try 'curl --help' or 'curl --manual' for more information
```

According to documentation I see there is nothing specified about different time units so if I didn't miss anything I suppose this is expected.  
I've tested then with something like `--rate 30\m` which is equivalent in a minute scale, but unfortunately the requests are pretty quick and 5 requests are made easily before the 10s frame. 

So is it possible to use custom time units?

Thank you!

**Comment by GlassGruber** [comment]
wow  sorry actually `--rate 30\m` _is_ working, can't tell what happened earlier the test file returned a forbidden response so I thought I was limited for too many requests in a short time. 

I see now that it has fractioned the batch to 1 request every 2 seconds, this interestingly showed me that the server can take a bit more time to respond so in 10s sometimes only 3 or 4 requests are made. 

So I guess it's needed  a dedicated script to do something like "run 5 requests in parallel and wait 10s for next batch" instead of a native curl function right?

Thank you!

**Reply by bagder** [reply, marked as answer]
I presume you mean `--rate 30/m`, because it needs to be a forward slash. This is also what the manpage documents: https://curl.se/docs/manpage.html#--rate

> So I guess it's needed a dedicated script to do something like "run 5 requests in parallel and wait 10s for next batch" instead of a native curl function right?

Right, until we enhance the functionality to also take a number of units as you tried first... I'll take a look at that.

**Reply by bagder** [reply]
See #14245

**Reply by GlassGruber** [reply]
> I presume you mean `--rate 30/m`, because it needs to be a forward slash. This is also what the manpage documents: https://curl.se/docs/manpage.html#--rate

Yeah you are totally right about the slash, it's a typo. 🙃 

> > So I guess it's needed a dedicated script to do something like "run 5 requests in parallel and wait 10s for next batch" instead of a native curl function right?
> 
> Right, until we enhance the functionality to also take a number of units as you tried first... I'll take a look at that.
>
> See https://github.com/curl/curl/pull/14245

This is great!  Thank you for your time answering back and for considering this issue/feature request!! 🎉🎉
